lms/djangoapps/teams/services.py

```diff
@@ -0,0 +1,20 @@
+""" Services to expose the Teams API to XBlocks """
+from __future__ import absolute_import
+
+from django.urls import reverse
+
+
+class TeamsService(object):
+    """ Functions to provide teams functionality to XBlocks"""
+    def get_team(self, user, course_id):
+        from . import api
+        return api.get_team_for_user_and_course(user, course_id)
+
+    def get_team_detail_url(self, team):
+        teams_dashboard_url = reverse('teams_dashboard', kwargs={'course_id': team.course_id})
+        # Unfortunately required since this URL resolution is done in a Backbone view
```

Comment:
I don't quite understand this - does it mean that we end up using this URL in a backbone view, and that view has not other way to calculate the URL outside of this method?

Reply:
The URL looks like this:
`edx.org/courses/blah/teams/#teams/topic_id/team_id`
up to the # is a django route. The remainder is passed into a backbone view and matched by a backbone router. I couldn't find anything else doing this "reverse"